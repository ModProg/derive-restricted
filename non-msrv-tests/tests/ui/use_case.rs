use derive_where::DeriveWhere;

#[derive(DeriveWhere)]
#[derive_where(Clone)]
struct StructEmpty {}

#[derive(DeriveWhere)]
#[derive_where(Clone)]
struct TupleEmpty();

#[derive(DeriveWhere)]
#[derive_where(Clone)]
struct Unit;

#[derive(DeriveWhere)]
#[derive_where(Clone)]
struct UnionEmpty {}

#[derive(DeriveWhere)]
#[derive_where(Clone)]
struct StructNone(u8);

#[derive(DeriveWhere)]
#[derive_where(Clone)]
enum EnumNone {
	A(u8),
}

#[derive(DeriveWhere)]
#[derive_where(Clone; T)]
struct SameGenerics<T>(T);

#[derive(DeriveWhere)]
#[derive_where(Clone, Debug; T)]
#[derive_where(skip_inner)]
struct Skip<T>(T);

#[derive(DeriveWhere)]
#[derive_where(Clone, Default; T)]
enum Default<T> {
	#[derive_where(default)]
	A(T),
}

fn main() {}
