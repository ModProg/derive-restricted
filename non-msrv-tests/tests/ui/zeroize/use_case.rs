extern crate zeroize_ as zeroize;

use derive_where::DeriveWhere;

#[derive(DeriveWhere)]
#[derive_where(Clone, Zeroize(crate = "zeroize_"); T)]
struct ZeroizeCrate<T>(T);

#[derive(DeriveWhere)]
#[derive_where(Clone, Zeroize; T)]
struct ZeroizeFqs<T>(#[derive_where(Zeroize(fqs))] T);

fn main() {}
