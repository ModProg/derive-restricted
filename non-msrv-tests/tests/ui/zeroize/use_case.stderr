error: this can be handled by standard `#[derive(..)]`, use a `skip` attribute, implement `Default` on an enum, of different generic type parameters
 --> tests/ui/zeroize/use_case.rs:6:16
  |
6 | #[derive_where(Clone, Zeroize(crate = "zeroize_"); T)]
  |                ^^^^^

error: this can be handled by standard `#[derive(..)]`, use a `skip` attribute, implement `Default` on an enum, of different generic type parameters
  --> tests/ui/zeroize/use_case.rs:10:16
   |
10 | #[derive_where(Clone, Zeroize; T)]
   |                ^^^^^
